crates/matrix-sdk/src/authentication/qrcode/rendezvous_channel.rs

```diff
@@ -0,0 +1,433 @@
+// Copyright 2024 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use http::{
+    header::{CONTENT_TYPE, ETAG, EXPIRES, IF_MATCH, IF_NONE_MATCH, LAST_MODIFIED},
+    HeaderMap, HeaderName, Method, StatusCode,
+};
+use ruma::api::{
+    client::error::ErrorKind,
+    error::{FromHttpResponseError, HeaderDeserializationError, IntoHttpError, MatrixError},
+    EndpointError,
+};
+use tracing::{debug, instrument, trace, warn};
+use url::Url;
+
+use crate::{http_client::HttpClient, HttpError, RumaApiError};
+
+const TEXT_PLAIN_CONTENT_TYPE: &str = "text/plain";
+const RECEIVE_TIMEOUT: Duration = Duration::from_secs(1);
```

Comment:
1s seems short for a low-end network. Do we want to increase it to 3s maybe?

Reply:
OK, after further read the code, I think we should rename this constant to `TIME_BEFORE_RETRY` or something like this?

Reply:
Would `POLL_TIMEOUT` work? The MSC uses this term:

![image](https://github.com/matrix-org/matrix-rust-sdk/assets/552026/9a36973f-aecd-407b-9a4c-8c07d1fdd2c4)

Reply:
Set it to `POLL_TIMEOUT` and set it to 1ms when run under tests, so our tests don't take seconds to run.